bip-xxxx.mediawiki

```diff
@@ -0,0 +1,283 @@
+     +-------+      PSBT        +------>+                      |
+     |       +------------------+       |                      |
+     |                                  |   +--------------+   |
+     |                                  |---+ Payjoin      |   |
+     |                                  |   | transaction  +-->+
+     |                                  |   +--------------+   |
+     +                                  +                      +
+</pre>
+The original PSBT is sent in the HTTP POST request body, base64 serialized, with <code>text/plain</code> in the <code>Content-Type</code> HTTP header and <code>Content-Length</code> set correctly.
+The payjoin proposal PSBT is sent in the HTTP response body, base64 serialized with HTTP code 200.
+
+===Receiver's well known errors===
+
+If for some reason the receiver is unable to create a payjoin proposal, it will reply with a HTTP code different than 200.
+The receiver is not constrained to specific set of errors, some are specified in this proposal.
+
+The errors have the following format:
+<pre>
+{
+    "errorCode": "leaking-data",
+    "message": "Key path information or GlobalXPubs should not be included in the original PSBT."
+}
+</pre>
+
+The well-known error codes are:
+{| class="wikitable"
+!Error code
+!Meaning
+|-
+|leaking-data
+|Key path information or GlobalXPubs should not be included in the original PSBT.
+|-
+|psbt-not-finalized
+|The original PSBT must be finalized.
+|-
+|unavailable
+|The payjoin endpoint is not available for now.
+|-
+|out-of-utxos
+|The receiver does not have any UTXO to contribute in a payjoin proposal.
+|-
+|not-enough-money
+|The receiver added some inputs but could not bump the fee of the payjoin proposal.
+|-
+|insane-psbt
+|Some consistency check on the PSBT failed.
+|-
+|version-unsupported
+|This version of payjoin is not supported.
+|-
+|need-utxo-information
+|The witness UTXO or non witness UTXO is missing
+|}
+
+The receiver is allowed to return implementation specific errors which may assist the sender to diagnose any issue.
+
+However, it is important that error codes that are not well-known and that the message do not appear on the sender's software user interface.
+Such error code or message could be used maliciously to fish a non technical user.
+Instead those error or message can only appear in debug logs.
+
+It is advise to hard code the description of the error code into the sender's software.
+
+===Sender's payjoin proposal checklist===
+
+The sender should check the payjoin proposal before signing it to prevent a malicious receiver from stealing money.
+
+* Check that all the spent outpoints in the original PSBT still exists in the coinjoin PSBT.
+* Check that all the spent outpoints in the original PSBT does not have any partial signature.
+* Check that the receiver added at least one input.
+* Check that the receiver inputs type match the inputs type of the sender. (ie. both using P2SH-P2WPKH or both using P2WPKH)
```

Comment:
If you provide an original psbt with multiple inputs of multiple types, this should not be enforced.